alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/solr/sharding/SolrShardingMetrics.java

```diff
@@ -0,0 +1,86 @@
+package eu.xenit.alfred.telemetry.binder.solr.sharding;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+import org.alfresco.repo.index.shard.ShardInstance;
+import org.alfresco.repo.index.shard.ShardRegistry;
+import org.alfresco.repo.index.shard.ShardRegistryImpl.ReplicaState;
+import org.alfresco.repo.index.shard.ShardState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SolrShardingMetrics {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SolrShardingMetrics.class);
+
+    public static final String SOLR_SHARDING_METRICS_PREFIX = "solr.sharding";
+
+    private SolrShardingMetricsContainer solrShardingMetricsContainer;
+    private MeterRegistry registry;
+
+    public SolrShardingMetrics(ShardRegistry shardRegistry, MeterRegistry registry) {
+        this.solrShardingMetricsContainer = new SolrShardingMetricsContainer(shardRegistry);
+        this.registry = registry;
+    }
+
+    private Map<String, AtomicLong> metrics = new HashMap<>();
+
+    public void updateMetrics() {
+        LOGGER.debug("Updating metrics");
+        solrShardingMetricsContainer.refresh();
+        solrShardingMetricsContainer.getFlocs().forEach(floc -> {
+            final Tags flocTags = Tags.of("floc", String.valueOf(floc.hashCode()))
+                    .and(floc.getStoreRefs().stream().map(storeRef -> Tag
+                            .of("storeRef", String.format("%s_%s", storeRef.getProtocol(), storeRef.getIdentifier())))
+                            .collect(Collectors.toSet()));
+            setAndCreateMetricIfNotExists("numberOfShards", floc.getNumberOfShards(), flocTags);
```

Comment:
As a best practice, the base unit should NOT be in the metric name, but should be provided to the metric with the `.baseUnit(string)` api call.

The reason is that some systems - like prometheus - want to use a base-unit-suffix. This metric would end up in Prometheus as as `solr_sharding_numberOfShards_number` or `solr_sharding_numberOfShards_total`

Reply:
where should I use that .baseUnit(string) call?

Reply:
where you create the `Gauge` - but that can be added later, this is fine!